config.yaml

```diff
@@ -11,8 +11,14 @@ options:
     default: master
   git_rev:
     type: string
-    description: the git revision (tag or hash) to check out
+    description: The git revision (tag or hash) to check out
     default: HEAD
+  git_depth:
+    type: int
+    description: >
+      Cloning depth, to truncate commit history to the specified number of commits.
+      Zero (default) means no truncating.
+    default: 0
```

Comment:
Any reason we are not defaulting to 1 here? Seems like it would make the most sense in the most situations.

Reply:
Confirmed with manual testing that sync will timeout and fail when offline.
Seems like the only advantage to keep a default of zero is fast switching, after history has been initially cloned once.
What do you think?